### PR TITLE
allow title pattern to be applied

### LINKED ISSUE #13394: TitlePart pattern not applied when options is set to 'Title is editable'
From titlePart settings of the TitlePart is editable and there is a pattern to be set the pattern does not get applied.

Solution .
Apply pattern   as long as the titelpart has not been set already.

### PR DESCRIPTION
**Pull request by giannik**
fixes #13394

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Title/Handlers/TitlePartHandler.cs [hunk 1]
```diff
@@ -64,8 +64,8 @@ private async Task SetTitleAsync(TitlePart part)
 
             var settings = GetSettings(part);
 
-            // Do not compute the title if the user can modify it.
-            if (settings.Options == TitlePartOptions.Editable || settings.Options == TitlePartOptions.EditableRequired)
+            // Do not compute the title if the user can modify it and the text is already set.
+            if ((settings.Options == TitlePartOptions.Editable || settings.Options == TitlePartOptions.EditableRequired)  && !String.IsNullOrWhiteSpace(part.ContentItem.DisplayText))
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Title/Handlers/TitlePartHandler.cs:68]: What you are suggesting looks like we had before #12712.

But this is now contradictory with the below condition line 70 (also added recently by #12712) which would never be true, and then line 74 would never be executed.

**Comment by giannik** [comment]
So how would you apply a pattern for a titlePart when its newly created ? Setting it to editable means you can edit it but  at least have the pattern apply if the value does'nt exist yet (eg for new content items)

**Comment by MikeAlhayek** [comment]
There was a PR that was merged about a week ago that would ensure the pattern is evaluated when an item is created. However, that logic won't bet executed if you make the option editable. When the title is editable you are saying that the user will provide the title. But when you say it's generated, the handler will create it during create or update.

Look how we are update the title when the option is not editable.

https://github.com/OrchardCMS/OrchardCore/blob/929cb9c1f6e731c24063127ce75fc1d5c21c9f1b/src/OrchardCore.Modules/OrchardCore.Title/Handlers/TitlePartHandler.cs#L41

**Comment by giannik** [comment]
So you cannot have an editable Titlepart that is initially autogenerated from the pattern. 
Should'nt these settings be simplified  like the settings in autoroute ?
So you may have something that is autogenerated initially but you can change it later on (like the slug in autoroute).

**Pull request closed without merging**